### ISSUE TITLE
Config data is re-encrypted when saved in the back office on Magento 2210

### ISSUE THREAD

**Issue by pixmin** [magento/magento2#12581]
### Preconditions
1. Magento 2.4-develop

### Steps to reproduce
1. Create a module with this a back office form, with such a field (system.xml):
```
[...]
<field id="hmackey" translate="label comment" type="text" sortOrder="50" showInDefault="1"
       showInWebsite="1" showInStore="1">
    <label>HMAC</label>
    <comment><![CDATA[...]]></comment>
    <validate>required-entry</validate>
    <config_path>foo/bar/hmackey</config_path>
    <backend_model>Magento\Config\Model\Config\Backend\Encrypted</backend_model>
</field>
[...]
```

2. Go to the admin on the form page
3. The data is correctly shown, save the form

### Expected result
1. Data is correctly encrypted and saved in the database
2. De-crypted data is shown in the admin form

### Actual result
1. Data is correctly encrypted and saved in the database
2. EN-crypted data is shown in the admin form

### Notes
I've tried the same configuration on previous versions and can't reproduce this issue, only on 2210.
I can't see anything in the changelog between 2200 and 2210 that would explain such a behavior.

Any ideas?

**Comment by pixmin** [comment]
G1 Failed? What does that mean? Did I fail to submit the issue correctly?

**Comment by pixmin** [comment]
I've updated the issue after it failed G1.

**Comment by magento-engcom-team** [comment]
Hi @pixmin 
Thanks for reporting this issue. 
This is not a bug.
For encrypted field you should specify attribute `type="obscure"`, like this:
`<field id="hmackey" translate="label comment" type="obscure" ...`

**Comment by pixmin** [comment]
Thank you for your prompt answer, but I do not want to obscure the field in the form, it has to be visible by the end user, as it needs to be read/verified/compared, but I do not want either to save it in clear in the database.

The configuration I use was working fine with previous Magento versions, up to 2200. It stopped working (and started re-encrypted the data) with version 2210.

**Comment by magento-engcom-team** [comment]
Hi @pixmin 
I understand you.
The fix for this issue is already available in the develop branch.

**Comment by pixmin** [comment]
I see you added a label "Fixed in 2.3.x", I'll check that now, it's still an issue in 2.2.2-dev.

Thanks again for your help!

**Comment by pixmin** [comment]
Update: Using Magento 2.3.0-dev solves this issue but creates a new one:
1. The values using the Encrypted backend model are **NOT** loaded from the configuration file (fields are empty)
2. Once entered by hand, they are correctly saved (not re-encrypted)

Config from etc/adminhtml/system.xml:
<backend_model>Magento\Config\Model\Config\Backend\Encrypted</backend_model>

**Comment by pixmin** [comment]
Just to confirm, removing the Encrypted Backend model from the field's configuration in `etc/adminhtml/system.xml` allows the value to be shown correctly in the back office, but it's (obviously) not encrypted anymore when saved in the database.

How is this fixed in 2.3.x?

**Comment by pixmin** [comment]
@okobchenko Hi, I see you removed the 2.2.x label, does that mean a fix has been applied to this version? Thanks :)

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-nazar. Thank you for working on this issue. 
Looks like this issue is already verified and confirmed. But if your want to validate it one more time, please, go though the following instruction:
- [x] 1. Add/Edit `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.

- [x] 2. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [x] 3. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 4. If the issue is not relevant or is not reproducible any more, feel free to close it.

**Comment by magento-engcom-team** [comment]
:white_check_mark: Confirmed by @engcom-Delta 
 Thank you for verifying the issue. Based on the provided information internal tickets `MC-30349` were created

**Issue Available**: @engcom-Delta, _You will be automatically unassigned. Contributors/Maintainers can claim this issue to continue. To reclaim and continue work, reassign the ticket to yourself._

**Comment by engcom-Charlie** [comment]
Hello @pixmin please try the following
1. Create `etc/adminhtml/system.xml` for example
```
    <system>
        <section id="sales">
            <group id="general">
                <field id="keyid" type="text" sortOrder="100" showInDefault="1" showInWebsite="1" showInStore="0">
                    <label>Key Id</label>
                    <backend_model>Magento\Config\Model\Config\Backend\Encrypted</backend_model>
                </field>
            </group>
        </section>
    </system>
```
2. Create `etc/config.xml`
```
    <default>
        <sales>
            <general>
                <keyid backend_model="Magento\Config\Model\Config\Backend\Encrypted" />
            </general>
        </sales>
    </default>
```

### AR:
we can see decrypted value in the admin form field, and in DB we can see the encrypted value.
![image](https://user-images.githubusercontent.com/51680850/103647635-49e4fe00-4f64-11eb-9dc0-d42db9a9ca35.png)
![image](https://user-images.githubusercontent.com/51680850/103647659-51a4a280-4f64-11eb-8726-f099a1ac2044.png)

Could you please take a look?
Thank you.

**Comment by engcom-Charlie** [comment]
Hello @pixmin, I'm closing this issue now due to inactivity.
Thank you for your collaboration.

**Comment by magento-engcom-team** [comment]
Hi @pixmin, @engcom-Charlie. 

Thank you for your report and collaboration!

The related internal Jira ticket `MC-30349` was closed as `Fixed`. 

The fix will be available with the upcoming `2.4.3` release.
